Chemistry/tests/SamplesTests/DocsSecondQuantization.cs

```diff
@@ -0,0 +1,340 @@
+            var spinOrbital = new SpinOrbital(5, Spin.d);
+
+            // Let us set the total number of orbitals to be say, `7`.
+            var nOrbitals = 7;
+
+            // This converts a spin-orbital index to a unique integer, in this case `12`,
+            // using the formula `g(j,σ)`.
+            var integerIndexHalfUp = spinOrbital.ToInt(IndexConvention.HalfUp, nOrbitals);
+
+            // This converts a spin-orbital index to a unique integer, in this case `11`,
+            // using the formula `h(j,σ)`.
+            var integerIndexUpDown = spinOrbital.ToInt(IndexConvention.UpDown);
+
+            // The default conversion uses the formula `h(j,σ)`, in this case `11`.
+            var integerIndexDefault = spinOrbital.ToInt();
+
+            Assert.True(integerIndexDefault == 11);
+
+            Assert.True(integerIndexHalfUp == 12);
+        }
+
+        [Fact]
+        static void LadderOperator()
+        {
+            // Let us use the spin orbital created in the previous snippet.
+            var spinOrbitalInteger = new SpinOrbital(5, Spin.d).ToInt();
+
+            // We specify either a creation or annihilation operator using 
+            // the enumerable type `RaisingLowering.u` or `RaisingLowering.d`
+            // respectively;
+            var creationEnum = RaisingLowering.u;
+
+            // The type representing a creation operator is then initialized 
+            // as follows. Here, we index these operators with integers.
+            // Hence we initialize the generic ladder operator with an
+            // integer index type.
+            var ladderOperator0 = new LadderOperator<int>(creationEnum, spinOrbitalInteger);
+
+            // An alternate constructor for a LadderOperator instead uses
+            // a tuple.
+            var ladderOperator1 = new LadderOperator<int>((creationEnum, spinOrbitalInteger));
+
+            Assert.True(ladderOperator0 == ladderOperator1);
+        }
+
+        [Fact]
+        static void FermionTerms()
+        {
+            // Let us initialize an array of tuples representing the
+            // desired sequence of creation operators.
+            var indices = new[] { (RaisingLowering.u, 1), (RaisingLowering.u, 2) };
+
+            // We can convert this array of tuples to a sequence of ladder
+            // operators using the `ToLadderSequence()` methods.
+            var ladderSequences = indices.ToLadderSequence();
+
+            // Sequences of ladder operators are quite general. For instance,
+            // they could be bosonic operators, intead of fermionic operators.
+            // We specialize them by constructing a `FermionTerm` representing 
+            // a fermion creation operator on the index `2` followed by `1`.
+            var fermionTerm = new FermionTerm(ladderSequences);
+
+            Assert.True(fermionTerm.Sequence.SequenceEqual(ladderSequences.Sequence));
+            Assert.True(fermionTerm.Coefficient == 1);
```

Comment:
In general, `Assert.Equal` provides more helpful diagnostic information in the case that the assertion fails.
```suggestion
            Assert.Equal(fermionTerm.Coefficient, 1);
```